Updater: Cannot find module 'typed-emitter'

<!--
issue_labeler_regex_version=0
--!>

<!-- Which version of electron-builder are you using? -->
<!-- Please always try to use latest version before reporting any issue. -->
* **Electron-Builder Version**: 23.0.9

<!-- Which version of node are you using? -->
* **Node Version**: 16

<!-- Which version of electron are you using? -->
<!-- Please verify if the error is electron related before reporting any issues. -->
* **Electron Version**: 16.2.8
* **Electron Type (current, beta, nightly)**: current
<!-- Which version of electron-updater are you using (if applicable)? --> 5.0.4

<!-- For which target are you building for? -->
* **Target**:  Windows

<!-- Enter your issue details below this comment. -->
<!-- If you want, you can donate to increase issue priority (https://www.electron.build/donate) -->

Unfortunately the latest update which adds types to AppUpdater to events does not work, the error is as follows:

```
node_modules/electron-updater/out/AppUpdater.d.ts:11:31 - error TS2307: Cannot find module 'typed-emitter' or its corresponding type declarations.
11 import type TypedEmitter from "typed-emitter";
```

I believe this is because typed-emitter is in devDependencies, instead of dependencies so it doesnt get downloaded

--- a/.changeset/purple-impalas-prove.md
+++ b/.changeset/purple-impalas-prove.md
@@ -1,0 +1,5 @@
+---
+"electron-updater": patch
+---
+
+fix: moving typed-emitter from devDependency to dependencies

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -3,7 +3,7 @@ version: 2
 jobs:
   build:
     docker:
-      - image: circleci/node:14.19
+      - image: cimg/node:14.19
     steps:
       - checkout
       - restore_cache:

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -13,7 +13,7 @@ jobs:
           keys:
             - v-11.0.0-electron
       - run:
-          command: npm i --prefix=$HOME/.local -g pnpm@6.25.1
+          command: npm i --prefix=$HOME/.local -g pnpm@7.1.6
       - run:
           command: pnpm install --frozen-lockfile
       - run:

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -3,7 +3,7 @@ version: 2
 jobs:
   build:
     docker:
-      - image: circleci/node:14.17@sha256:b1c0b133fc872314dee5b24e34084693f0747850aad6ac0181b70061e198375f
+      - image: circleci/node:14.19
     steps:
       - checkout
       - restore_cache:

--- a/.github/actions/pnpm/action.yml
+++ b/.github/actions/pnpm/action.yml
@@ -4,7 +4,7 @@ inputs:
   version: # id of input
     description: 'The pnpm version to use'
     required: false
-    default: 6.25.1
+    default: 7.1.6
 
 runs:
   using: 'composite'

--- a/.github/actions/pnpm/action.yml
+++ b/.github/actions/pnpm/action.yml
@@ -9,16 +9,18 @@ inputs:
 runs:
   using: 'composite'
   steps:
-    - uses: pnpm/action-setup@v2
-      with:
-        version: ${{ inputs.version }}
-
-    - uses: actions/setup-node@v3
+    - name: Setup node
+      uses: actions/setup-node@v3
       with:
         node-version: '14'
         cache: 'pnpm'
 
-    - name: install dependencies
+    - name: Setup pnpm
+      uses: pnpm/action-setup@v2
+      with:
+        version: ${{ inputs.version }}
+
+    - name: Install dependencies
       run: pnpm install --frozen-lockfile
       shell: bash
 

--- a/.github/actions/pnpm/action.yml
+++ b/.github/actions/pnpm/action.yml
@@ -10,7 +10,7 @@ runs:
   using: 'composite'
   steps:
     - name: Setup pnpm
-      uses: pnpm/action-setup@v2
+      uses: pnpm/action-setup@v2.2.2
       with:
         version: ${{ inputs.version }}
 

--- a/.github/actions/pnpm/action.yml
+++ b/.github/actions/pnpm/action.yml
@@ -9,16 +9,16 @@ inputs:
 runs:
   using: 'composite'
   steps:
+    - name: Setup pnpm
+      uses: pnpm/action-setup@v2
+      with:
+        version: ${{ inputs.version }}
+
     - name: Setup node
       uses: actions/setup-node@v3
       with:
         node-version: '14'
         cache: 'pnpm'
-
-    - name: Setup pnpm
-      uses: pnpm/action-setup@v2
-      with:
-        version: ${{ inputs.version }}
 
     - name: Install dependencies
       run: pnpm install --frozen-lockfile

--- a/.github/actions/pretest/action.yml
+++ b/.github/actions/pretest/action.yml
@@ -15,16 +15,16 @@ inputs:
 runs:
   using: 'composite'
   steps:
+    - name: Setup pnpm
+      uses: pnpm/action-setup@v2
+      with:
+        version: ${{ inputs.version }}
+
     - name: Setup node
       uses: actions/setup-node@v3
       with:
         node-version: '14'
         cache: 'pnpm'
-
-    - name: Setup pnpm
-      uses: pnpm/action-setup@v2
-      with:
-        version: ${{ inputs.version }}
 
     - name: Install dependencies
       run: pnpm install --frozen-lockfile

--- a/.github/actions/pretest/action.yml
+++ b/.github/actions/pretest/action.yml
@@ -16,7 +16,7 @@ runs:
   using: 'composite'
   steps:
     - name: Setup pnpm
-      uses: pnpm/action-setup@v2
+      uses: pnpm/action-setup@v2.2.2
       with:
         version: ${{ inputs.version }}
 

--- a/.github/actions/pretest/action.yml
+++ b/.github/actions/pretest/action.yml
@@ -7,20 +7,26 @@ inputs:
   cache-path:
     description: 'The path to the electron cache'
     required: true
+  version:
+    description: 'The pnpm version to use'
+    required: false
+    default: 7.1.6
 
 runs:
   using: 'composite'
   steps:
-    - uses: pnpm/action-setup@v2
-      with:
-        version: 7.1.6
-
-    - uses: actions/setup-node@v3
+    - name: Setup node
+      uses: actions/setup-node@v3
       with:
         node-version: '14'
         cache: 'pnpm'
 
-    - name: install dependencies
+    - name: Setup pnpm
+      uses: pnpm/action-setup@v2
+      with:
+        version: ${{ inputs.version }}
+
+    - name: Install dependencies
       run: pnpm install --frozen-lockfile
       shell: bash
 

--- a/.github/actions/pretest/action.yml
+++ b/.github/actions/pretest/action.yml
@@ -13,7 +13,7 @@ runs:
   steps:
     - uses: pnpm/action-setup@v2
       with:
-        version: 6.25.1
+        version: 7.1.6
 
     - uses: actions/setup-node@v3
       with:

--- a/docker/build.sh
+++ b/docker/build.sh
@@ -7,7 +7,7 @@ DATE=$(date +%m.%y)
 docker build -t electronuserland/builder:base -t "electronuserland/builder:base-$DATE" docker/base
 
 # Node 14
-docker build --build-arg NODE_VERSION=14.19.1 -t electronuserland/builder:14 -t "electronuserland/builder:14-$DATE" -t electronuserland/builder:latest docker/node
+docker build --build-arg NODE_VERSION=14.19.3 -t electronuserland/builder:14 -t "electronuserland/builder:14-$DATE" -t electronuserland/builder:latest docker/node
 
 docker build --build-arg IMAGE_VERSION=14 -t electronuserland/builder:14-wine -t "electronuserland/builder:14-wine-$DATE" -t electronuserland/builder:wine docker/wine
 docker build --build-arg IMAGE_VERSION=14-wine -t electronuserland/builder:14-wine-mono -t "electronuserland/builder:14-wine-mono-$DATE" -t electronuserland/builder:wine-mono docker/wine-mono

--- a/docker/node/Dockerfile
+++ b/docker/node/Dockerfile
@@ -1,6 +1,6 @@
 FROM electronuserland/builder:base
 
-ARG NODE_VERSION 14.17.0
+ARG NODE_VERSION 14.19.3
 
 # this package is used for snapcraft and we should not clear apt list - to avoid apt-get update during snap build
 RUN curl -L https://nodejs.org/dist/v$NODE_VERSION/node-v$NODE_VERSION-linux-x64.tar.gz | tar xz -C /usr/local --strip-components=1 && \

--- a/docker/node/Dockerfile
+++ b/docker/node/Dockerfile
@@ -7,4 +7,4 @@ RUN curl -L https://nodejs.org/dist/v$NODE_VERSION/node-v$NODE_VERSION-linux-x64
   unlink /usr/local/CHANGELOG.md && unlink /usr/local/LICENSE && unlink /usr/local/README.md && \
   # https://github.com/npm/npm/issues/4531
   npm config set unsafe-perm true
-RUN npm i -g pnpm@6.25.1
+RUN npm i -g pnpm@7.1.6

--- a/packages/app-builder-lib/src/targets/MsiTarget.ts
+++ b/packages/app-builder-lib/src/targets/MsiTarget.ts
@@ -224,7 +224,9 @@ export default class MsiTarget extends Target {
         result += `>\n`
         const shortcutName = commonOptions.shortcutName
         if (isCreateDesktopShortcut) {
-          result += `${fileSpace}  <Shortcut Id="desktopShortcut" Directory="DesktopFolder" Name="${xmlAttr(shortcutName)}" WorkingDirectory="APPLICATIONFOLDER" Advertise="yes" Icon="${this.iconId}"/>\n`
+          result += `${fileSpace}  <Shortcut Id="desktopShortcut" Directory="DesktopFolder" Name="${xmlAttr(
+            shortcutName
+          )}" WorkingDirectory="APPLICATIONFOLDER" Advertise="yes" Icon="${this.iconId}"/>\n`
         }
 
         const hasMenuCategory = commonOptions.menuCategory != null
@@ -233,7 +235,9 @@ export default class MsiTarget extends Target {
           if (hasMenuCategory) {
             dirs.push(`<Directory Id="${startMenuShortcutDirectoryId}" Name="ProgramMenuFolder:\\${commonOptions.menuCategory}\\"/>`)
           }
-          result += `${fileSpace}  <Shortcut Id="startMenuShortcut" Directory="${startMenuShortcutDirectoryId}" Name="${xmlAttr(shortcutName)}" WorkingDirectory="APPLICATIONFOLDER" Advertise="yes" Icon="${this.iconId}">\n`
+          result += `${fileSpace}  <Shortcut Id="startMenuShortcut" Directory="${startMenuShortcutDirectoryId}" Name="${xmlAttr(
+            shortcutName
+          )}" WorkingDirectory="APPLICATIONFOLDER" Advertise="yes" Icon="${this.iconId}">\n`
           result += `${fileSpace}    <ShortcutProperty Key="System.AppUserModel.ID" Value="${xmlAttr(this.packager.appInfo.id)}"/>\n`
           result += `${fileSpace}  </Shortcut>\n`
         }
@@ -275,10 +279,5 @@ function listToString(list: Array<string>, indentLevel: number) {
 }
 
 function xmlAttr(str: string) {
-  return str
-    .replace(/&/g, "&amp;")
-    .replace(/</g, "&lt;")
-    .replace(/>/g, "&gt;")
-    .replace(/"/g, "&quot;")
-    .replace(/'/g, "&apos;")
+  return str.replace(/&/g, "&amp;").replace(/</g, "&lt;").replace(/>/g, "&gt;").replace(/"/g, "&quot;").replace(/'/g, "&apos;")
 }

--- a/packages/electron-updater/package.json
+++ b/packages/electron-updater/package.json
@@ -23,14 +23,14 @@
     "lazy-val": "^1.0.5",
     "lodash.escaperegexp": "^4.1.2",
     "lodash.isequal": "^4.5.0",
-    "semver": "^7.3.5"
+    "semver": "^7.3.5",
+    "typed-emitter": "^2.1.0"
   },
   "devDependencies": {
     "@types/fs-extra": "9.0.13",
     "@types/js-yaml": "4.0.3",
     "@types/lodash.escaperegexp": "4.1.6",
-    "@types/lodash.isequal": "4.5.5",
-    "typed-emitter": "^2.1.0"
+    "@types/lodash.isequal": "4.5.5"
   },
   "typings": "./out/main.d.ts",
   "publishConfig": {

